File: src/main/kotlin/com/akolov/currigate/AuthClient.java
package com.akolov.currigate;


import java.util.List;

public class AuthClient {
    private String clientId;

    private String clientSecret;
    private String accessTokenUri;
    private String userAuthorizationUri;
    private String userInfoUri;
    private List<String> scope;
    private String clientAuthenticationScheme;

    public String getUserInfoUri() {
        return userInfoUri;
    }

    public void setUserInfoUri(String userInfoUri) {
        this.userInfoUri = userInfoUri;
    }


    public String getClientId() {
        return clientId;
    }

    public String getClientSecret() {
        return clientSecret;
    }

    public String getAccessTokenUri() {
        return accessTokenUri;
    }

    public String getUserAuthorizationUri() {
        return userAuthorizationUri;
    }

    public String getClientAuthenticationScheme() {
        return clientAuthenticationScheme;
    }

    public List<String> getScope() {
        return scope;
    }

    public void setScope(List<String> scope) {
        this.scope = scope;
    }

    public void setClientId(String clientId) {
        this.clientId = clientId;
    }

    public void setClientSecret(String clientSecret) {
        this.clientSecret = clientSecret;
    }

    public void setAccessTokenUri(String accessTokenUri) {
        this.accessTokenUri = accessTokenUri;
    }

    public void setUserAuthorizationUri(String userAuthorizationUri) {
        this.userAuthorizationUri = userAuthorizationUri;
    }

    public void setClientAuthenticationScheme(String clientAuthenticationScheme) {
        this.clientAuthenticationScheme = clientAuthenticationScheme;
    }
}
                      

File: src/main/kotlin/com/akolov/currigate/AuthClient.kt
package com.akolov.currigate

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component
import java.util.*


@Component
class AuthClient {

    @Value("\${oauth2.google.client.clientId}")
    var clientId: String? = null
    @Value("\${oauth2.google.client.secret}")
    var clientSecret: String? = null
    @Value("\${oauth2.google.client.accessTokenUri}")
    var accessTokenUri: String? = null
    @Value("\${oauth2.google.client.userAuthorizationUri}")
    var userAuthorizationUri: String? = null
    @Value("\${oauth2.google.client.userInfoUri}")
    var userInfoUri: String? = null

    val scope: List<String> = Arrays.asList("openid","email", "profile")

}




